beraborrow.go

```diff
@@ -0,0 +1,204 @@
+// InfraredWrappers are 1-1 with the CompoundingInfraredCollateralVault token type
+
+// LPTDecimals are the decimals of the InfraredWrapper token
+// CDPDecimals are the decimals of the CompoundingInfraredCollateralVault token
+type BeraBorrowCDPConfig struct {
+	ColVaultAddress string `json:"col_vault_address"`
+	LPTDecimals     uint   `json:"lpt_decimals"`
+	CDPDecimals     uint   `json:"cdp_decimals"`
+}
+
+// BeraBorrowLPPriceProvider defines the provider for BeraBorrow CDP price and CDP TVL.
+type BeraBorrowLPPriceProvider struct {
+	LPTAddress  common.Address
+	logger      zerolog.Logger
+	block       *big.Int
+	configBytes []byte
+	config      *BeraBorrowCDPConfig
+	lptContract *sc.BeraBorrowIW
+	cdpContract *sc.BeraBorrowCICV
+}
+
+// NewBeraBorrowLPPriceProvider creates a new instance of the BeraBorrowLPPriceProvider.
+func NewBeraBorrowLPPriceProvider(LPTAddress common.Address, block *big.Int, logger zerolog.Logger, config []byte) *BeraBorrowLPPriceProvider {
+	b := &BeraBorrowLPPriceProvider{
+		LPTAddress:  LPTAddress,
+		logger:      logger,
+		block:       block,
+		configBytes: config,
+	}
+	return b
+}
+
+// Initialize checks the configuration/data and instantiates the CDP contract.
+func (b *BeraBorrowLPPriceProvider) Initialize(ctx context.Context, client *ethclient.Client) error {
+	var err error
+
+	b.config = &BeraBorrowCDPConfig{}
+	err = json.Unmarshal(b.configBytes, b.config)
+	if err != nil {
+		b.logger.Error().Err(err).Msg("failed to deserialize config")
+		return err
+	}
+
+	// Initialize the IW contract from the address in the inputs for the LP token address
+	b.lptContract, err = sc.NewBeraBorrowIW(b.LPTAddress, client)
+	if err != nil {
+		b.logger.Error().Err(err).Msg("adapter init failed to instatiate Infrared Wrapper contract")
+		return err
+	}
+
+	// Initialize the CICV contract from the address in the config -- not the IW contract from the LP token address
+	b.cdpContract, err = sc.NewBeraBorrowCICV(common.HexToAddress(b.config.ColVaultAddress), client)
+	if err != nil {
+		b.logger.Error().Err(err).Msg("adapter init failed to instatiate Compounding Infrared Collateral Vault contract")
+		return err
+	}
+
+	return nil
+}
+
+// LPTokenPrice returns the current price of the protocol's LP token in USD
+func (b *BeraBorrowLPPriceProvider) LPTokenPrice(ctx context.Context) (string, error) {
+
+	opts := &bind.CallOpts{
+		Pending:     false,
+		Context:     ctx,
+		BlockNumber: b.block,
+	}
+
+	tvl, err := b.cicvTotalValue(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	// returns a *big.Int for total supply of the LP token
+	lpTotalSupply, err := b.lptContract.BeraBorrowIWCaller.TotalSupply(opts)
+	if err != nil {
+		return "", err
+	}
+	numTokens := NormalizeAmount(lpTotalSupply, b.config.LPTDecimals)
+	pricePerToken := tvl.Div(numTokens)
+
+	b.logger.Info().
+		Str("pricePerToken", pricePerToken.String()).
+		Msg("LP token price calculated successfully")
+
+	return pricePerToken.StringFixed(8), nil
```

Comment:
As we discussed, all numbers you use should be moved to constants with meaningful names and comments.